crates/provider/src/ext/tenderly.rs

```diff
@@ -0,0 +1,203 @@
+//! This module extends the Ethereum JSON-RPC provider with the Tenderly namespace's RPC methods.
+use crate::Provider;
+use alloy_eips::BlockNumberOrTag;
+use alloy_network::Network;
+use alloy_primitives::TxHash;
+use alloy_rpc_types_eth::{state::StateOverride, BlockOverrides};
+use alloy_rpc_types_tenderly::TenderlySimulationResult;
+use alloy_transport::TransportResult;
+
+/// Tenderly namespace rpc interface that gives access to several non-standard RPC methods.
+///
+/// Some methods are currently not implemented:
+/// - tenderly_estimateGas
+/// - tenderly_gasPrice
+/// - tenderly_suggestGasFee
+/// - tenderly_estimateGasBundle
+/// - tenderly_decodeInput
+/// - tenderly_decodeError
+/// - tenderly_decodeEvent
+/// - tenderly_functionSignatures
+/// - tenderly_errorSignatures
+/// - tenderly_eventSignatures
+/// - tenderly_getTransactionRange
+/// - tenderly_getContractAbi
+/// - tenderly_getStorageChanges
+#[cfg_attr(target_family = "wasm", async_trait::async_trait(?Send))]
+#[cfg_attr(not(target_family = "wasm"), async_trait::async_trait)]
+pub trait TenderlyApi<N: Network>: Send + Sync {
+    /// Simulates a transaction as it would execute on the given block, allowing overrides of state
+    /// variables and balances of all accounts
+    async fn tenderly_simulate_transaction(
+        &self,
+        tx: N::TransactionRequest,
+        block: BlockNumberOrTag,
+        state_overrides: Option<StateOverride>,
+        block_overrides: Option<BlockOverrides>,
```

Comment:
to make this a bit nice I suggest we do something like

https://github.com/alloy-rs/alloy/blob/d9ec782cc1e3a7850f44f72884aea8f6b9b11ca8/crates/provider/src/provider/eth_call/mod.rs#L147-L158

here, I think we could even reuse this type here?